### ISSUE TITLE
Report an error if an invalid parameter is used on a filter, or an invalid filter is used

### ISSUE THREAD

**Issue by thesunshade** [suttacentral/suttacentral#2978]
#### User story

Sven wants to search just in the Theravada Abhidhamma. Logically, it would make sense for `in:in:pli-tv-ab` to work, but it doesn't. However there is no explicit indication to Sven when he tries it. He just gets no results.

Natalia wants to search just in Ven. Suddhaso's translations, so she adds `author:suddaso`. She gets no results, but does not realize that it's because she spelled his name wrong.

Diệp wants to limit her results to Bhante Sujato, so she adds `authour:sujato`. She gets no results, but does not realize that it's because she spelled the filter wrong.

#### Feature description

Two error reporting types should happen for filters:

1. When a filter takes only a set of parameters, if the user tries an invalid parameter they should be told, for example:
```
pli-tv-ab is not a valid parameter for `in:` please check the spelling or bla bla bla…
```
2. When someone tries to use a filter that does not exist, they should be told, for example:

```
The filter `authour` does not exist. Check the spelling or refer to the list of parameters.
```
An extra bonus would be if the filter has a list option they could be told something additionally like:

```
To get a list of all authors for your selected languages, search `list authors`
```

#### Pre milestone planning check:
- [ ] Small enough to completed in a milestone.
- [ ] Dependencies marked
- [ ] No external dependencies block the PBI from being completed.
- [ ] Details are understood by dev team to decide if the PBI can be completed.

#### Done check:
- [ ] Produced code for presumed functionalities
- [ ] Project builds without errors
- [ ] Peer Code Review performed/pull request approved x2
- [ ] Project deployed on the stage environment identical to production platform
- [ ] Feature is tested against acceptance criteria
- [ ] Feature ok-ed by Product Owner (moving to Closed on the Board)
- [ ] Refactoring completed
- [ ] Any configuration or build changes documented (readme, etc...)

**Comment by khemarato** [comment]
Yes. The SERP should definitely let people know if search can't parse their query.

**Comment by thesunshade** [comment]
> if search can't parse their query.

I wonder if there are other circumstances that could lead to the query not being parsed.

**Comment by thesunshade** [comment]
@ihongda and @buddhist-uni, I'm wondering how easy you think this will be to implement. I ask because Bhante has given the ok for https://github.com/suttacentral/suttacentral/issues/2970 to change `author:` to `by:`. But I thought it might be better to implement the error checking first in case folks have old links to searches that use this filter.

**Comment by ihongda** [comment]
> [@ihongda](https://github.com/ihongda) and [@buddhist-uni](https://github.com/buddhist-uni), I'm wondering how easy you think this will be to implement. I ask because Bhante has given the ok for [#2970](https://github.com/suttacentral/suttacentral/issues/2970) to change `author:` to `by:`. But I thought it might be better to implement the error checking first in case folks have old links to searches that use this filter.

@thesunshade 

Currently I keep both `author` and `by` filters.

I tried to fix this issue and pushed it to `staging`

**Comment by thesunshade** [comment]
This works for me!

Ideally there would be a link that opened up the filter overlay so people could understand the filters better. Not sure if that is possible.

**Comment by thesunshade** [comment]
Once we have this pushed to production, I suggest that we remove `author` as a filter.

as a side note, `author:sujato` and `by:sujato` give different looking results:

![Image](https://github.com/user-attachments/assets/16aa9834-b205-48f9-a05f-6adf84a4b16a)

![Image](https://github.com/user-attachments/assets/de972c56-bac9-48be-89af-cc4ba428025c)

I'm not sure which is better.

**Comment by thesunshade** [comment]
Sorry for all the messages…

You may know, but the error checking is only for incorrect filters, not the parameter. So for example https://staging.suttacentral.net/search?query=author:suddaso returns a blank page. Suddhaso is spelled wrong.

![Image](https://github.com/user-attachments/assets/ed230981-4405-4ffe-8346-d32b203fd3ea)
